planner/core/testdata/integration_suite_out.json

```diff
@@ -3605,310 +3605,294 @@
       {
         "SQL": "explain format = 'brief' select count(*) from fact_t, d1_t where fact_t.d1_k = d1_t.d1_k",
         "Plan": [
-          "HashAgg 1.00 root  funcs:count(Column#12)->Column#11",
-          "└─TableReader 1.00 root  data:ExchangeSender",
-          "  └─ExchangeSender 1.00 mpp[tiflash]  ExchangeType: PassThrough",
-          "    └─HashAgg 1.00 mpp[tiflash]  funcs:count(1)->Column#12",
-          "      └─HashJoin 32.00 mpp[tiflash]  inner join, equal:[eq(test.d1_t.d1_k, test.fact_t.d1_k)]",
-          "        ├─ExchangeReceiver(Build) 4.00 mpp[tiflash]  ",
-          "        │ └─ExchangeSender 4.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.d1_t.d1_k, collate: binary]",
-          "        │   └─Selection 4.00 mpp[tiflash]  not(isnull(test.d1_t.d1_k))",
-          "        │     └─TableFullScan 4.00 mpp[tiflash] table:d1_t keep order:false",
-          "        └─ExchangeReceiver(Probe) 16.00 mpp[tiflash]  ",
-          "          └─ExchangeSender 16.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.fact_t.d1_k, collate: binary]",
-          "            └─Selection 16.00 mpp[tiflash]  not(isnull(test.fact_t.d1_k))",
-          "              └─TableFullScan 16.00 mpp[tiflash] table:fact_t keep order:false"
+          "StreamAgg 1.00 root  funcs:count(1)->Column#11",
+          "└─TableReader 32.00 root  data:ExchangeSender",
+          "  └─ExchangeSender 32.00 mpp[tiflash]  ExchangeType: PassThrough",
+          "    └─HashJoin 32.00 mpp[tiflash]  inner join, equal:[eq(test.d1_t.d1_k, test.fact_t.d1_k)]",
+          "      ├─ExchangeReceiver(Build) 4.00 mpp[tiflash]  ",
+          "      │ └─ExchangeSender 4.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.d1_t.d1_k, collate: binary]",
+          "      │   └─Selection 4.00 mpp[tiflash]  not(isnull(test.d1_t.d1_k))",
+          "      │     └─TableFullScan 4.00 mpp[tiflash] table:d1_t keep order:false",
+          "      └─ExchangeReceiver(Probe) 16.00 mpp[tiflash]  ",
+          "        └─ExchangeSender 16.00 mpp[tiflash]  ExchangeType: HashPartition, Hash Cols: [name: test.fact_t.d1_k, collate: binary]",
+          "          └─Selection 16.00 mpp[tiflash]  not(isnull(test.fact_t.d1_k))",
+          "            └─TableFullScan 16.00 mpp[tiflash] table:fact_t keep order:false"
         ]
       },
       {
         "SQL": "explain format = 'brief' select count(*) from fact_t, d1_t, d2_t, d3_t where fact_t.d1_k = d1_t.d1_k and fact_t.d2_k = d2_t.d2_k and fact_t.d3_k = d3_t.d3_k",
         "Plan": [
-          "HashAgg 1.00 root  funcs:count(Column#18)->Column#17",
```

Comment:
Expected, 1) StreamAgg is better than HashAgg if no much data to process; 2) no need to push the Agg down since there are only 128 rows.